src/components/bookmark-list-container.ts

```diff
@@ -288,60 +142,28 @@ export class BookmarkListContainer extends LitElement {
   };
 
   private handleSyncRequested = async () => {
-    if (this.containerState.isSyncing) return;
-
-    try {
-      const settings = await DatabaseService.getSettings();
-      if (settings) {
-        await SyncService.syncBookmarks(settings);
-        // Don't dispatch sync-requested event to avoid infinite loop
-        // The sync service itself will dispatch the necessary events
-      }
-    } catch (error) {
-      console.error('Failed to sync bookmarks:', error);
-    }
+    await this.syncController.requestSync();
   };
 
   private handleFaviconLoadRequested = async (bookmarkId: number, faviconUrl: string) => {
-    try {
-      const faviconDataUrl = await FaviconService.getFaviconForBookmark(bookmarkId, faviconUrl);
-      
-      // Update cache
-      const updatedCache = new Map(this.containerState.faviconCache);
-      updatedCache.set(bookmarkId, faviconDataUrl);
-      
-      this.containerState = {
-        ...this.containerState,
-        faviconCache: updatedCache,
-      };
-    } catch (error) {
-      console.error('Failed to load favicon:', error);
-    }
+    await this.faviconController.loadFavicon(bookmarkId, faviconUrl);
   };
 
   private handleVisibilityChanged = (visibleBookmarkIds: number[]) => {
-    // Load favicons for visible bookmarks
-    visibleBookmarkIds.forEach(bookmarkId => {
-      const bookmark = this.containerState.bookmarks.find(b => b.id === bookmarkId);
-      if (bookmark && bookmark.favicon_url && !this.containerState.faviconCache.has(bookmarkId)) {
-        this.handleFaviconLoadRequested(bookmarkId, bookmark.favicon_url);
-      }
-    });
+    this.faviconController.handleVisibilityChanged(visibleBookmarkIds, this.containerState.bookmarks);
   };
 
   override render() {
+    const syncState = this.syncController.getSyncState();
+    const faviconState = this.faviconController.getFaviconState();
+    
     return html`
       <bookmark-list
         .bookmarks=${this.containerState.bookmarks}
         .isLoading=${this.containerState.isLoading}
-        .syncState=${{
-          isSyncing: this.containerState.isSyncing,
-          syncProgress: this.containerState.syncProgress,
-          syncTotal: this.containerState.syncTotal,
-          syncedBookmarkIds: this.containerState.syncedBookmarkIds,
-        }}
+        .syncState=${syncState}
```

Comment:
It looks like the BookmarkList mainly uses `syncState` to render a sync progress bar at the top of the list. This isn't directly related to displaying a list of bookmarks (the purpose of BookmarkList) and visually it's also separate from the rest of the BookmarkList UI. Let's move the sync progress bar UI into BookmarkListContainer. That way the BookmarkList doesn't need `syncState` at all and only needs the list of recently-synced bookmarks.